### ISSUE TITLE
Is the producer a tcp keepalive connection

### ISSUE THREAD

**Issue by yanmxa** [confluentinc/confluent-kafka-go#1191]
Description
===========

Is the producer client a keepalive connection? Cause I found some error from the producer eventChan: ' 2/2 all brokers are down`. Is that mean the producer keeps connection with the brokers all the time?
Or just connect to the brokers sending messages and after sending messages then shut down the connection?

**Comment by jmathai-confluent** [comment]
Yes, the producer maintains persistent TCP connections to brokers. The 'all brokers are down' error indicates a connectivity issue, not normal behavior. Connection keepalive is controlled by `socket.keepalive.enable` and `connections.max.idle.ms` configuration. If you have a specific issue, please open a new discussion.
